public/app/plugins/panel/table/TablePanel.tsx

```diff
@@ -91,21 +105,184 @@ export class TablePanel extends Component<Props> {
+      return this.getSummaryFooterValues(frame);
+    }
+
+    return undefined;
+  }
+
+  getValues = (): FieldDisplay[] => {
+    const { data, options, replaceVariables, fieldConfig, timeZone } = this.props;
+    const currentIndex = this.getCurrentFrameIndex(data.series);
+    return getFieldDisplayValues({
+      fieldConfig,
+      reduceOptions: options.reduceOptions || { calcs: [] },
+      replaceVariables,
+      theme: config.theme2,
+      data: [data.series[currentIndex]],
+      timeZone,
+    });
+  };
+
+  getSummaryFooterValues(frame: DataFrame) {
+    const { options } = this.props;
+
+    const summary = options.reduceOptions;
+    if (summary === undefined) {
+      return [];
+    }
+    if (Array.isArray(summary.calcs) && summary.calcs.length === 0) {
+      return [];
+    }
+
+    const fields = options.footerSummary?.fields;
+    // calculate the cells
+    const cells = frame.fields.map((f, i) => {
+      if (fields || fields === '') {
+        if (fields.includes('') && f.type === FieldType.number) {
+          return this.calculateSummaryCell(i);
+        }
+        if (fields.includes(f.name)) {
+          return this.calculateSummaryCell(i);
+        }
+        if (i === 0 && summary.calcs && summary.calcs.length > 0) {
+          const reducer = summary.calcs[0];
+          const formatted = this.capitalizeFirstLetter(reducer.replace(/([A-Z])/g, ' $1').trim());
+          return formatted;
+        }
+      }
+      return [];
+    });
+
+    // if there are no multi-value cells, don't show the label in the cell
+    // it will be shown on the first column
+    const multiValue = cells.find((cell) => Array.isArray(cell) && cell.length > 1);
+    if (!multiValue) {
+      const reducer = summary.calcs[0];
+      const formatted = this.capitalizeFirstLetter(reducer.replace(/([A-Z])/g, ' $1').trim());
+      const clean = cells.map((cell, i) => {
+        if (Array.isArray(cell) && cell.length > 0) {
+          const cellValue = cell[0];
+          const key: string = Object.keys(cellValue)[0];
+          return cellValue[key];
+        }
+        if (i === 0 && cell === '') {
+          return formatted;
+        }
+        return cell;
+      });
+      return clean;
+    }
+    return cells;
+  }
+
+  capitalizeFirstLetter(value: string) {
+    return value.charAt(0).toUpperCase() + value.slice(1);
+  }
+
+  // get the footer cells from data provided by another frame (pre-calculated)
+  getProvidedFooterValues(frame: DataFrame): FooterItem[] | undefined {
+    const { options, data } = this.props;
+
+    const summaryFrame = data.series.find((f) => f.name === options.footerFrame);
+    if (summaryFrame === undefined) {
+      return undefined;
+    }
+
+    // group cells by field for multi value cells
+    const cellsByField: KeyValue = {};
+    frame.fields.forEach((f, i) => {
+      for (const sf of summaryFrame!.fields) {
+        const referenceField: string = sf.config?.custom?.['referenceField'];
```

Comment:
@ryantxu this is currently what I'm using to link the totals field/column to the details field/column.  If this is ok, seems we don't need to select a frame for the footer, just use the frame with this custom meta.

Reply:
I don't really get what this is for -- is this to give us the same formatting?

Is it ever something different than the corresponding column?

Reply:
In salesforce the summary field name is "s!AMOUNT" (sum of AMOUNT) which is a reference to AMOUNT.  Could also have a field called a!AMOUNT (avg of AMOUNT).  If I strip the prefix I may have 2 fields with the same name.

Reply:
Summary Frame:
```
| Sum of Amount | Avg of Amount | 
----------------------------------
        500            200                      
```

Footer ( 2 fields from the summary frame reference one field in the detail frame and are stacked)
```
| Amount |
----------
 sum: 500
 avg: 200

```